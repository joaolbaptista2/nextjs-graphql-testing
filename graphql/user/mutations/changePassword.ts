import gql from "graphql-tag";

export const changePasswordMutation = gql`
  mutation ChangePassword($data: ChangePasswordInput!) {
    changePassword(data: $data) {
      id,
      name,
      lastName,
      email
    }
}
`;