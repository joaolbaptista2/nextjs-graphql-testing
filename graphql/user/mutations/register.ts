import gql from "graphql-tag";

export const loginMutation = gql`
mutation Register($data: RegisterInput!) {
  register(data: $data) {
    id
    firstName
    lastName
    email
    name
  }
}
`;