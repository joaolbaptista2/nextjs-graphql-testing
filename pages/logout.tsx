import { MyContext } from '../interfaces/MyContext'
import { LogoutMutation } from '../generated/apolloComponents'
import { logoutMutation } from '../graphql/user/mutations/logout'
import redirect from '../lib/redirect';
const Logout = () => {
  return null;
}

Logout.getInitialProps = async ({apolloClient, ...ctx}: MyContext) => {
  await apolloClient.mutate<LogoutMutation>({
    mutation: logoutMutation
  })
  await apolloClient.resetStore()
  redirect(ctx, '/login')
  return {}
}

export default Logout;