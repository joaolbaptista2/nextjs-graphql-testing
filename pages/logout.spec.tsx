import React from 'react'
import renderer from 'react-test-renderer'
import { MockedProvider, wait } from '@apollo/react-testing';
import { logoutMutation } from '../graphql/user/mutations/logout'
import Logout from './logout';
import Router  from 'next/router';

jest.mock('next/router', () => ({
  replace: () => {}
}))
let mockReplace;
beforeEach(() => {
 mockReplace = jest.spyOn(Router, 'replace');
})
afterEach(() => {
 jest.clearAllMocks()
})

describe('<Logout />', () => {
  const tree = renderer.create(<Logout />)
  it('should not regress', () => {
    expect(tree.toJSON).toMatchSnapshot()
  })

  it('should return null', () => {
    const instance = tree.root
    expect(instance.children).toMatchObject([])
  })

  it('should call logout mutation on getInitialProps', async () => {
    // To be finished
    const mocks = [
      {
        request: {
          query: logoutMutation
        },
        newData: jest.fn(() => ({
          data: {
            logout: true
            }
        })),
      },
    ]
    const logoutMockResult = mocks[0].newData;
    await renderer.create(
      <MockedProvider mocks={mocks} addTypename={false}>
        <Logout />
      </MockedProvider>
    )

    const props = await Logout.getInitialProps({apolloClient: { mutate: jest.fn(), resetStore: jest.fn()}, redirect: jest.fn()} as any)
    expect(mockReplace).toBeCalled()
    expect(props).toEqual({})
  })
})
