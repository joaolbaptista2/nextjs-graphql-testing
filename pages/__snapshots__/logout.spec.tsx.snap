// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logout /> should not regress 1`] = `[Function]`;
